hail/src/main/scala/is/hail/linalg/BlockMatrix.scala

```diff
@@ -1546,6 +1441,62 @@ private class BlockMatrixMultiplyRDD(l: BlockMatrix, r: BlockMatrix)
   @transient override val partitioner: Option[Partitioner] = Some(gp)
 }
 
+case class BlockMatrixRectanglesRDD(rectangles: Array[Array[Long]], bm: BlockMatrix)
+  extends RDD[(Int, BDM[Double])](bm.blocks.sparkContext, Nil) {
+
+  assert(rectangles.forall(rect => rect.length == 4))
+
+  val gp: GridPartitioner = bm.gp
+
+  override def compute(split: Partition, context: TaskContext): Iterator[(Int, BDM[Double])] = {
+    val rect = rectangles(split.index)
+    val rectStartRow = rect(0)
+    val rectEndRow = rect(1)
+    val rectStartCol = rect(2)
+    val rectEndCol = rect(3)
```

Comment:
This could alternatively be:
```scala
val Array(rectStartRow, rectEndRow, rectStartCol, rectEndCol) = rect
```

I dunno if that's more or less clear. It will helpfully throw an error if we have incorrectly sized arrays.

Reply:
I like that better. Fewer magic numbers and clearly shows the reasoning behind the structure of the array.